### PR TITLE
WIP: Errors in logs should show log tags as well.

### LINKED ISSUE #22979: Request IDs don't show on every line for errors
Requests tagged with `:request_id` do not show up on ever line when there is an error. Example: 

```
[7f8f84ce-d134-4b77-88dc-bac10b54a410]
2016-01-08T20:38:23.835130+00:00 app[web.3]: ActionView::Template::Error (undefined method `new_user_registration_path' for #<ActionDispatch::Routing::RoutesProxy:0x007fece81b1548>):
2016-01-08T20:38:23.835131+00:00 app[web.3]:     3: <% end -%>
2016-01-08T20:38:23.835132+00:00 app[web.3]:     4:
2016-01-08T20:38:23.835133+00:00 app[web.3]:     5: <%- if devise_mapping.registerable? && controller_name != 'registrations' %>
2016-01-08T20:38:23.835134+00:00 app[web.3]:     6:   <%= link_to "Sign up", new_registration_path(resource_name) %><br />
2016-01-08T20:38:23.835134+00:00 app[web.3]:     7: <% end -%>
2016-01-08T20:38:23.835135+00:00 app[web.3]:     8:
2016-01-08T20:38:23.835136+00:00 app[web.3]:     9: <%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
2016-01-08T20:38:23.835137+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.5/lib/action_dispatch/routing/routes_proxy.rb:36:in `method_missing'
2016-01-08T20:38:23.835138+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/devise-3.5.2/lib/devise/controllers/url_helpers.rb:51:in `block (4 levels) in generate_helpers!'
2016-01-08T20:38:23.835139+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/devise-3.5.2/app/views/devise/shared/_links.html.erb:6:in `_vendor_bundle_ruby_______gems_devise_______app_views_devise_shared__links_html_erb__2183085242633923340_70327838930360'
2016-01-08T20:38:23.835140+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/template.rb:145:in `block in render'
2016-01-08T20:38:23.835141+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.5/lib/active_support/notifications.rb:166:in `instrument'
2016-01-08T20:38:23.835142+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/template.rb:333:in `instrument'
2016-01-08T20:38:23.835143+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/template.rb:143:in `render'
2016-01-08T20:38:23.835144+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/renderer/partial_renderer.rb:339:in `render_partial'
2016-01-08T20:38:23.835144+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/renderer/partial_renderer.rb:310:in `block in render'
2016-01-08T20:38:23.835145+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'
2016-01-08T20:38:23.835146+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.5/lib/active_support/notifications.rb:164:in `block in instrument'
2016-01-08T20:38:23.835147+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.5/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
2016-01-08T20:38:23.835147+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.5/lib/active_support/notifications.rb:164:in `instrument'
2016-01-08T20:38:23.835148+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'
2016-01-08T20:38:23.835149+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/renderer/partial_renderer.rb:309:in `render'
2016-01-08T20:38:23.835150+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/renderer/renderer.rb:47:in `render_partial'
2016-01-08T20:38:23.835150+00:00 app[web.3]:   vendor/bundle/ruby/2.3.0/gems/actionview-4.2.5/lib/action_view/helpers/rendering_helper.rb:35:in `render'
```

versus

```
2016-01-08T20:38:28.672073+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b] Processing by PagesController#index as HTML
2016-01-08T20:38:28.685667+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b]   Rendered application/_github.html.slim (0.2ms)
2016-01-08T20:38:28.758460+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b]   Rendered repos/_repo.html.slim (39.8ms)
2016-01-08T20:38:28.762702+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b]   Rendered pages/index.html.slim within layouts/application (84.9ms)
2016-01-08T20:38:28.765791+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b]   Rendered application/_flashes.html.slim (0.2ms)
2016-01-08T20:38:28.770927+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b]   Rendered application/_nav.html.slim (4.7ms)
2016-01-08T20:38:28.776329+00:00 app[web.1]: [547856d7-3f24-484c-8f40-0cd9b9d26c2b]   Rendered application/_footer.html.slim (4.1ms)
```

### PR DESCRIPTION
**Pull request by vipulnsward**
- Changed formatted_code_for to return array of logs to be tagged for each line
- Changed some render tests to match new behaviour of return

Fixes #22979 

r? @schneems 

Opening to get feedback on the logger to be used. We construct new logger from env, that doesn't log tags, which for example- `ActionView::Base.logger` does.
 What's an appropriate logger here though? I believe `ActionView::Base.logger`, since thats where the error arises from.

### CODE REVIEW

**Comment by schneems** [comment]
Sorry for taking so long, this looks good. Can you take a look at failing tests?

```
  1) Failure:
DebugExceptionsTest#test_uses_logger_from_env [/home/travis/build/rails/rails/actionpack/test/dispatch/debug_exceptions_test.rb:351]:
Expected /puke/ to match "".
  2) Failure:
DebugExceptionsTest#test_logs_exception_backtrace_when_all_lines_silenced [/home/travis/build/rails/rails/actionpack/test/dispatch/debug_exceptions_test.rb:374]:
Expected 0 to be > 10.
```

**Comment by vipulnsward** [comment]
@schneems all green now.

**Comment by schneems** [comment]
:rocket:  thanks!

File: actionpack/lib/action_dispatch/middleware/debug_exceptions.rb [hunk 1]
```diff
@@ -149,22 +149,26 @@ def render(status, body, format)
     def log_error(request, wrapper)
       logger = logger(request)
       return unless logger
-
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/debug_exceptions.rb]: :cry: Why was this removed?

File: actionpack/lib/action_dispatch/middleware/debug_exceptions.rb [hunk 2]
```diff
@@ -149,22 +149,26 @@ def render(status, body, format)
...
+        logger.fatal "#{exception.class} (#{exception.message}):"
+        log_array logger, exception.annoted_source_code if exception.respond_to?(:annoted_source_code)
+        logger.fatal "  "
+        log_array logger, trace
       end
     end
 
-    def logger(request)
-      request.logger || stderr_logger
+    def log_array logger, array
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/debug_exceptions.rb]: Parentheses please :pray:
